src/WebJobs.Extensions.CosmosDB/CosmosDBAttribute.cs

```diff
@@ -102,6 +102,23 @@ public CosmosDBAttribute(string databaseName, string collectionName)
         [AutoResolve(ResolutionPolicyType = typeof(CosmosDBSqlResolutionPolicy))]
         public string SqlQuery { get; set; }
 
+        /// <summary>
+        /// Optional.
+        /// Enable to use with Multi Master accounts.
+        /// </summary>
+        public bool UseMultipleWriteLocations { get; set; }
+
+        /// <summary>
+        /// Optional.
+        /// Defines preferred locations (regions) for geo-replicated database accounts in the Azure Cosmos DB service.
+        /// Values should be comma-separated.
+        /// </summary>
+        /// <example>
+        /// PreferredLocations = "East US,South Central US,North Europe"
+        /// </example>
+        [AutoResolve]
+        public string PreferredLocations { get; set; }
+
```

Comment:
So now we have a mix of ConnectionPolicy values that we can set on either the global config (`ConnectionMode`, `ConnectionProtocol`) or the individual attributes (`UseMultipleWriteLocations`, `PreferredLocations`). Is that confusing? Should we support them all in both places, with attribute overriding config? I know the original issue was building up the cache with unique keys, but you have a way to do that now...

Reply:
The logic I thought was that ConnectionMode and ConnectionProtocol are environment configurations, they depend on where the Function is running since they are affected by corporate firewalls for example.
While the new attributes are particular to each binding, you can have 2 Output bindings with different PreferredLocations for example.

Reply:
Yeah, that's right. Thanks for reminding me.

Reply:
Yeah, that's right. Thanks for reminding me.